BootloaderCommonPkg/Library/ShellLib/CmdBoot.c

```diff
@@ -266,26 +266,38 @@ GetBootFileInfo (
   EFI_STATUS                 Status;
   UINTN                      Length;
 
-  do {
-    ShellPrint (L"Enter file path string (max length of %d)\n",
-                sizeof (BootOption->Image[0].FileName) - 1
-                );
-    ShellPrint (L"(default '%a') ", CurrOption->Image[0].FileName);
-    Status = ShellReadLine (Shell, Buffer, BufferSize);
-    if (EFI_ERROR (Status)) {
-      return Status;
-    }
-    Length = StrLen (Buffer);
-    if (Length == 0) {
-      CopyMem (BootOption->Image[0].FileName, CurrOption->Image[0].FileName, sizeof (CurrOption->Image[0].FileName));
-      break;
-    }
-    if (Length < sizeof (BootOption->Image[0].FileName)) {
-      UnicodeStrToAsciiStrS (Buffer, (CHAR8 *)BootOption->Image[0].FileName, sizeof (BootOption->Image[0].FileName));
-      break;
+    for (int image_index = 0; image_index < LoadImageTypeMax; image_index++){
+      if (CurrOption->Image[image_index].FileName[0] != '\0'){
+        do {
+          if (image_index == 0){
+            ShellPrint (L"Enter file path string (max length of %d)\n",
+                        sizeof (BootOption->Image[0].FileName) - 1
+                        );
+          }
+          else {
+            ShellPrint (L"Enter file path string for extra image %d (max length of %d)\n",
+                        image_index,
+                        sizeof (BootOption->Image[0].FileName) - 1
+                        );
+          }
+          ShellPrint (L"(default '%a') ", CurrOption->Image[image_index].FileName);
+          Status = ShellReadLine (Shell, Buffer, BufferSize);
+          if (EFI_ERROR (Status)) {
+            return Status;
+          }
+          Length = StrLen (Buffer);
+          if (Length == 0) {
+            CopyMem (BootOption->Image[image_index].FileName, CurrOption->Image[image_index].FileName, sizeof (CurrOption->Image[image_index].FileName));
+            break;
+          }
+          if (Length < sizeof (BootOption->Image[image_index].FileName)) {
+            UnicodeStrToAsciiStrS (Buffer, (CHAR8 *)BootOption->Image[image_index].FileName, sizeof (BootOption->Image[image_index].FileName));
+            break;
+          }
+          ShellPrint (L"Invalid, too long: '%s' len=%d, please re-enter\n", Buffer, Length);
+        } while (1);

```

Comment:
It is not clear for user the input file is for what by mixing them together. And let's skip Image[2] for now since it is for misc image and might need special handling.

Reply:
Noted and submitted patch